src/site/mod.rs

```diff
@@ -45,48 +45,60 @@ impl Site {
-                    pages.push(artifacts_page);
+        let mut index = Page::index(&layout_template, config)?;
+
+        if Self::needs_context(config) {
+            match &config.repository {
+                Some(repo_url) => {
+                let context = Context::new(repo_url, config.artifacts.cargo_dist)?;
+                if config.artifacts.has_some() {
+                    index = Page::index_with_artifacts(&context, &layout_template, config)?;
+                    if context.latest_dist_release.is_some()
+                        || config.artifacts.package_managers.is_some()
+                    {
+                        let body = artifacts::page(&context, config)?;
+                        let artifacts_page = Page::new_from_contents(
+                            body,
+                            "artifacts.html",
+                            &layout_template,
+                            config,
+                        );
+                        pages.push(artifacts_page);
+                    }
                 }
-            }
-            if config.changelog {
-                let changelog_html = changelog::build(&context, config)?;
-                let changelog_page = Page::new_from_contents(
-                    changelog_html,
-                    "changelog.html",
-                    &layout_template,
-                    config,
-                );
-                let changelog_releases = changelog::build_all(&context, config)?;
-                pages.push(changelog_page);
-                for (name, content) in changelog_releases {
-                    let page = Page::new_from_contents(
-                        content,
-                        &format!("changelog/{}.html", name),
+                if config.changelog {
+                    let changelog_html = changelog::build(&context, config)?;
+                    let changelog_page = Page::new_from_contents(
+                        changelog_html,
+                        "changelog.html",
                         &layout_template,
                         config,
                     );
-                    pages.push(page);
+                    let changelog_releases = changelog::build_all(&context, config)?;
+                    pages.push(changelog_page);
+                    for (name, content) in changelog_releases {
+                        let page = Page::new_from_contents(
+                            content,
+                            &format!("changelog/{}.html", name),
+                            &layout_template,
+                            config,
+                        );
+                        pages.push(page);
+                    }
                 }
+            },
+            None => Err(OrandaError::Other("You have indicated you want to use features that require a repository context. Please make sure you have a repo listed in your project or oranda config."))
```

Comment:
I would prefer that this error message specifically names the key to change, but I guess that's fuzzier with axoproject involved and merging results..? (also the second sentence should use a proper 'help')

Reply:
updated the message. going through and making sure we're using full miette style errors is gonna be a diff PR